cve_bin_tool/cli.py

```diff
@@ -103,15 +103,23 @@ def scan_file(self, filename):
             raise InvalidFileError(filename)
 
         # step 1: check if it's an ELF binary file
-        o = subprocess.check_output(['file', filename])
-        if sys.version_info.major == 3:
-            o = o.decode('utf-8')
-
-        if 'cannot open' in o:
-            raise InvalidFileError(filename)
-
-        if ("LSB " not in o) and ("LSB shared" not in o) and ("LSB executable" not in o) and ("PE32 executable" not in o) and ("PE32+ executable" not in o) and ("Mach-O" not in o):
-            return None
+        import binascii
+        with open(filename, 'rb') as f:
+            byte = f.read(4)
+            signature = binascii.hexlify(byte)
+            # little endian order
+            if sys.byteorder == 'little':
+                # ELF signature in unix
+                if signature.decode('utf-8') == '7f454c46':
+                    pass
+                else:
+                    return None
+            # big endian order
+            else:
+                if signature.decode('utf-8') == '457f464c':
```

Comment:
If I'm not mistaken. endianness does not apply to sequences of bytes, only to things like ints being loaded and stored from memory into/out of registers. I think we should leave this out, @terriko, thoughts?

Reply:
My pc is little endian, when I print the signature, it shows 7f454c46, which is the little endian order of the original ELF signature 457f464c. Thus I think endianness matters and added it to code.

Reply:
ok sounds good